prov/tcp/src/tcpx_progress.c

```diff
@@ -421,8 +421,13 @@ void tcpx_progress(struct util_ep *util_ep)
 
 	ep = container_of(util_ep, struct tcpx_ep, util_ep);
 	fastlock_acquire(&ep->lock);
+
+	if (ep->cm_state != TCPX_EP_CONNECTED)
+		goto out;
```

Comment:
This introduces a branch in the fast path.  Can the calling code just not call this function if we're in the wrong state?  Describing the call path that reaches here incorrect in the commit message would help.